grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -648,6 +648,16 @@ class OrderService {
                         productPackage.quantity == orderItem.quantityPerUom
             }
 
+            // If not found, then we look for a product package associated with the product
+            if (!productPackage && !orderItem?.productSupplier) {
+                productPackage = orderItem.product.packages.find { ProductPackage productPackage1 ->
```

Comment:
I don't like the name `productPackage1`, but nothing better comes to my mind, maybe other developers will have some ideas :thinking:

Reply:
it's been like this before, hence I assumed this would fit OK.

Reply:
I had the same reaction, but I've probably done this as well. This is a case where the default `it` would probably make it less confusing, but that's not an official convention so meh. 

In general, if there's a way to describe a variable in a more meaningful way, then do it. Here, since we're basically iterating through and trying to find a default product package, I would call it defaultProductPackage.

A better (more DRY, OOO) approach would be to add a convenience method to Product.

```
if (!productPackage && !orderItem?.productSupplier) {
                productPackage = orderItem.product.getDefaultProductPackage(orderItem.quantityPerUom, orderItem.quantityUom)
```
The actual method might look like this
```
// Does not handle if multiple product packages match (but this should happen due to unique constraint)
// Does not pass product since we make the call on orderItem.product
ProductPackage getDefaultProductPackage(Integer quantityPerUom, quantityUom) {
    return productPackages.find { ProductPackage productPackage ->
        return productPackage.uom == orderItem.quantityUom &&
            productPackage.quantity == orderItem.quantityPerUom &&
            !productPackage.productSupplier
    }
}
```

Reply:
@jmiranda would you like me to move it to a method now, or you just wanted to make a point for the future? 
If the answer is, that I should move it, then I think I'd also have to pass the `orderItem` as well (beside `quantityPerUom` and `uom`), because if we want to have it in the `Product` domain, we would not have the access to the `orderItem`.

Reply:
Yeah it was mostly a pattern to keep an eye out for in the future.

Reply:
As far as I can tell we don't need orderItem there. Could you explain the need for that?

Reply:
Sorry, I'm missing something here. What do you need from the order item that we're not already passing into the new method? 

> Oops. Sorry, I just realized I had a typo in my previous version of getDefaultProductPackage. I've now removed the "orderItem" from within the find closure. 
>
> ![image](https://github.com/openboxes/openboxes/assets/1162935/539ffd45-9e20-4052-bd2a-ba2289d2d043)


The new method on Product.groovy doesn't need the orderItem, it would take the quantity and uom as arguments. 
```
ProductPackage getDefaultProductPackage(Integer quantityPerUom, UnitOfMeasure quantityUom) { 
    return productPackages.find { ProductPackage productPackage -> 
        return productPackage.uom == quantityUom && 
            productPackage.quantity == quantityPerUom && 
            !productPackage.productSupplier } 
}
```
And then the caller in updateProductPackage would just invoke the default product package method using the properties from orderItem. 
```
    // If not found, then we look for a product package associated with the product
    if (!productPackage && !orderItem?.productSupplier) {
        productPackage = orderItem.product.getDefaultProductPackage(orderItem.quantityPerUom, orderItem.quantityUom)
        }
    }
```